config/application.rb

```diff
@@ -240,7 +240,7 @@ def cache_store_config
     config.three_scale.cors.enabled = false
     config.three_scale.cors.merge!(try_config_for(:cors) || {})
 
-    three_scale = config_for(:settings).symbolize_keys
+    three_scale = config_for(:settings).deep_symbolize_keys
```

Comment:
I don't quite get this one to be honest :upside_down_face: 

I think some things broke because of this change: https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#rails-application-config-for-return-value-no-longer-supports-access-with-string-keys

However, it says that what stopped being supported is access with string keys, but what I observe is that the difference is that now the hash that is produced by `config_for` on the contrary has string keys, and not symbols like it used to be. So, maybe I'm missing something... 

But this specific change fixes several issues with the `impersonation_admin` config.

Reply:
Yeah, but we are using the Rails `config_for`, it's just the description of the change is not 100% aligned with the actual behavior. But to be honest, I already don't remember how it used to behave, and how it behaves now. If needed, I can recheck :) 

But this works.

Reply:
> Yeah, but we are using the Rails `config_for`

If I place a breakpoint on our `config_for` it stops many times, so we use it. Anyway, if it works, fine for me.

Reply:
@jlledom Yes yes, we DO use this method, what I meant is we then use Rails' (via `super`) `config_for`.

I've been looking into it, and indeed, the Rails' description is correct. Rails' `config_for` now returns a config that is deep-symbolized already. But when we apply `.with_indifferent_access`, and then `.symbolize_keys` that was lost, and accessing the nested values by symbols didn't work: `Rails.application.config.three_scale.impersonation_admin[:domain]` was nil.

Applying `.deep_symbolize_keys` fixed this. But also, I think we transform the configs too much, and they are not consistent (`deep_symbolize_keys` is applied for some, `symbolize_keys` applied for others etc.).

So, I thought we could take this opportunity to align all, and to use the default format that Rails provides - `ActiveSupport::OrderedOptions` with deep-symbolized keys.
We just need to make sure we use symbols throughout the app.